File: app/src/main/java/com/my/televip/obfuscate/struct/ResolverRegistry.java
package com.my.televip.obfuscate.struct;

import com.my.televip.Clients.ClientManager;

public class ResolverRegistry {

    Class<?> finalParameter;
    Class<?> clazz;

    public ResolverRegistry(Class<?> clazz){
        this.clazz = clazz;
        try {
            Class<?> parameterResolverClass = null;
            for (Class<?> inner : clazz.getDeclaredClasses()) {
                if (inner.getSimpleName().equals("ParameterResolver")) parameterResolverClass = inner;
            }

            finalParameter = parameterResolverClass;

        } catch (Throwable ignored) {}
    }

    public boolean hasParameter(String name){
        try {
            return (boolean)finalParameter.getMethod("has", String.class).invoke(null, name);
        } catch (Throwable e){
            return false;
        }
    }

    public Class<?>[] resolveParameter(String name){
        try {
            return (Class<?>[]) finalParameter.getMethod("resolve", String.class).invoke(null, name);
        } catch (Throwable e){
            return null;
        }
    }

    public void loadParameter(){
        try {
            clazz.getMethod("loadParameter").invoke(null);
        } catch (Throwable ignored){}
    }

    public static Class<?> getResolverClass() {
        ClientManager.Client clientType = ClientManager.getCurrent();
        if (clientType == null) return null;
        return clientType.getResolverClass();
    }
}

File: app/src/main/java/com/my/televip/obfuscate/struct/ResolverRegistry.kt
package com.my.televip.obfuscate.struct

import com.my.televip.clients.ClientManager

class ResolverRegistry(private val clazz: Class<*>?) {

    private var finalParameter: Class<*>? = null

    init {
        try {
            if (clazz != null) {
                var parameterResolverClass: Class<*>? = null
                for (inner in clazz.declaredClasses) {
                    if (inner.simpleName == "ParameterResolver") {
                        parameterResolverClass = inner
                    }
                }
                finalParameter = parameterResolverClass
            }
        } catch (ignored: Throwable) {
        }
    }

    fun hasParameter(name: String): Boolean {
        return try {
            val paramClass = finalParameter ?: return false
            val method = paramClass.getMethod("has", String::class.java)
            (method.invoke(null, name) as? Boolean) ?: false
        } catch (e: Throwable) {
            false
        }
    }

    fun resolveParameter(name: String): Array<Class<*>>? {
        return try {
            val paramClass = finalParameter ?: return null
            val method = paramClass.getMethod("resolve", String::class.java)
            @Suppress("UNCHECKED_CAST")
            method.invoke(null, name) as? Array<Class<*>>
        } catch (e: Throwable) {
            null
        }
    }

    fun loadParameter() {
        try {
            clazz?.getMethod("loadParameter")?.invoke(null)
        } catch (ignored: Throwable) {
        }
    }

    companion object {
        @JvmStatic
        fun getResolverClass(): Class<*>? {
            val clientType = ClientManager.getCurrent() ?: return null
            return clientType.resolverClass
        }
    }
}
